tests/js/client/shell/shell-check-metrics-move-noncluster.js

```diff
@@ -0,0 +1,112 @@
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / @author Max Neunhoeffer
+// //////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require('jsunity');
+
+function getMetric(name) {
+  let res = arango.GET_RAW( '/_admin/metrics/v2');
+  if (res.code !== 200) {
+    throw "error fetching metric";
+  }
+  let re = new RegExp("^" + name);
+  let matches = res.body.split('\n').filter((line) => !line.match(/^#/)).filter((line) => line.match(re));
+  if (!matches.length) {
+    throw "Metric " + name + " not found";
+  }
+  return Number(matches[0].replace(/^.* (\d+)$/, '$1'));
+}
+
+function checkMetricsMoveSuite() {
+  'use strict';
+
+  return {
+    
+    testByProtocol: function () {
+      let http1ReqCount = getMetric("arangodb_request_body_size_http1_count");
+      let http1ReqSum = getMetric("arangodb_request_body_size_http1_sum");
+      let http2ReqCount = getMetric("arangodb_request_body_size_http2_count");
+      let http2ReqSum = getMetric("arangodb_request_body_size_http2_sum");
+      let vstReqCount = getMetric("arangodb_request_body_size_vst_count");
+      let vstReqSum = getMetric("arangodb_request_body_size_vst_sum");
+      // Do a few requests:
+      for (let i = 0; i < 10; ++i) {
+        let res = arango.GET_RAW("/_api/version");
+        assertEqual(200, res.code);
+      }
+      // Note that GET requests do not have bodies, so they do not move
+      // the sum of the histogram. So let's do a PUT just for good measure:
+      let logging = arango.GET("/_admin/log/level");
+      let res = arango.PUT_RAW("/_admin/log/level", logging);
+      assertEqual(200, res.code);
+      // And get the values again:
+      let http1ReqCount2 = getMetric("arangodb_request_body_size_http1_count");
+      let http1ReqSum2 = getMetric("arangodb_request_body_size_http1_sum");
+      let http2ReqCount2 = getMetric("arangodb_request_body_size_http2_count");
+      let http2ReqSum2 = getMetric("arangodb_request_body_size_http2_sum");
+      let vstReqCount2 = getMetric("arangodb_request_body_size_vst_count");
+      let vstReqSum2 = getMetric("arangodb_request_body_size_vst_sum");
+      if (arango.protocol() === "vst") {
+        assertNotEqual(vstReqCount, vstReqCount2);
+        assertNotEqual(vstReqSum, vstReqSum2);
+      } else {
+        assertEqual(vstReqCount, vstReqCount2);
+        assertEqual(vstReqSum, vstReqSum2);
+      }
+      if (arango.protocol() === "http") {
+        assertNotEqual(http1ReqCount, http1ReqCount2);
+        assertNotEqual(http1ReqSum, http1ReqSum2);
+      } else {
+        assertEqual(http1ReqCount, http1ReqCount2);
+        assertEqual(http1ReqSum, http1ReqSum2);
+      }
```

Comment:
If the protocol used during the test is "http", we will count our own requests (executed by this test) plus all requests that are sent to the server by other tasks (or internally). I think this only works because the test is named "-noncluster". If this test would run in cluster, too, all the cluster-internal requests would be counted as well and the test wouldn't be deterministic. 
So I guess it works, but we should add a comment to the test that this works on in non-cluster mode because otherwise cluster-internal requests would be counted.

Reply:
Right, I will add a comment.

Reply:
Done.